plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -214,6 +215,7 @@ private synchronized void releaseCachedClients() {
         }
         // clear previously cached clients, they will be build lazily
         clientsCache = emptyMap();
+        derivedClientSettings = emptyMap();
```

Comment:
The logic of just blank clearing out everything here isn't great but with the complicated way in which we handle the AWS references I figured I'd rather not make this code more complicated and for now keep invalidating the whole cache on closing any repo.